xbmc/platform/android/CPUInfoAndroid.cpp

```diff
@@ -8,17 +8,43 @@
 
 #include "CPUInfoAndroid.h"
 
+#include "URL.h"
+#include "utils/StringUtils.h"
 #include "utils/Temperature.h"
 
 #include "platform/android/activity/AndroidFeatures.h"
 
+#include <array>
+
 std::shared_ptr<CCPUInfo> CCPUInfo::GetCPUInfo()
 {
   return std::make_shared<CCPUInfoAndroid>();
 }
 
 CCPUInfoAndroid::CCPUInfoAndroid()
 {
+  m_pPosixFile = std::make_unique<CPosixFile>();
```

Comment:
Any reason for storing the file pointer as a member? Isn't it only used in the scope of this method?
If so wouldn't a local variable be better?

Reply:
The reason was to reuse the same `CPosixFile` instance since `GetCPUFrequency` is called periodically at regular intervals while user is in System info screen.

Although seems interval is long as 15 seconds so no much difference.